### DISCUSSION TITLE
Error 62 (0x3e): GATT CONN FAIL ESTABLISH; ESP32 C3

### DISCUSSION THREAD

**Discussion by ficwiktor** [h2zero/NimBLE-Arduino#674] [Q&A]
Hi, 
I am using Seeeduino Xiao ESP32 C3 and none of the Server examples seem to work for me. There are no compilation errors, device is discoverable but each connection fails. 

Screenshots after using NimBLE_Secure_Server.ino example:
![441877867_449863267627843_7026898777922677998_n](https://github.com/h2zero/NimBLE-Arduino/assets/52634139/064364cb-528f-439b-8d12-35bfaaeef17c)

![441940569_2136004660099189_5733646289122362259_n](https://github.com/h2zero/NimBLE-Arduino/assets/52634139/8773b8a0-c7c9-46f6-b170-db86b484f6e9)

and on pc, using bleak python library:
![Zrzut ekranu 2024-05-31 092120](https://github.com/h2zero/NimBLE-Arduino/assets/52634139/ef37d22c-653e-4353-9ab3-b788c24945a2)

Other examples yield same results.

![Zrzut ekranu 2024-05-31 093031](https://github.com/h2zero/NimBLE-Arduino/assets/52634139/7e7730ce-9388-41f8-adca-8740595aaeff)

I couldn't find anything in Issues or in Discussions.

**Comment by h2zero** [comment]
Hello, this can be caused by bad xtal accuracy on the board, please try another board and see if you have the same issue.

**Reply by lukewakeford** [reply]
Hey, @h2zero  I just had this same issue, another board from the same manufacturer worked just fine, thanks for the tip 👍🏻.
Is there anything I can do to fix the boards with bad xtal accuracy? or are the going in the bin?

Thanks.

**Reply by h2zero** [reply]
Unless you can replace the xtal then no, but you can use them for non-ble stuff.

**Reply by lukewakeford** [reply]
Ah i see, many thanks.
